### PR TITLE
 PUBDEV-8637 - row to tree assignments implemented model.rowToTreeAssignment API

### PR DESCRIPTION
**Pull request by valenad1**
https://h2oai.atlassian.net/browse/PUBDEV-8637

Hello, here is the implementation for rowToTreeAssignment output from the GBM model. Java implementation is here.

- Python https://github.com/h2oai/h2o-3/pull/6465
- R https://github.com/h2oai/h2o-3/pull/6466

### CODE REVIEW

File: h2o-algos/src/test/java/hex/tree/gbm/GBMRowToTreeAssignmentTest.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    public void setup() {
+        final File h2oHomeDir = new File(System.getProperty("user.dir")).getParentFile();
+        environmentVariables.set("H2O_FILES_SEARCH_PATH", h2oHomeDir.getAbsolutePath());
+    }
+
+    @Test
+    public void testRowToTreeAssignmentSmokeBinomialHardcoded() {
+        Scope.enter();
+        try {
+            Scope.enter();
```
Comment (mn-mikke) [inline h2o-algos/src/test/java/hex/tree/gbm/GBMRowToTreeAssignmentTest.java:45]: Do we need to enter the scope multiple times?

Comment (valenad1) [reply h2o-algos/src/test/java/hex/tree/gbm/GBMRowToTreeAssignmentTest.java:45]: No, fixed 👍

**Review by mn-mikke — APPROVED** [review]
LGTM, thank you a lot for your time explaining me the details!

**Review by wendycwong — APPROVED** [review]
Nice!

**Pull request merged**